### PR TITLE
[serverless] add http.status_code to all inferred spans as default

### PR DESCRIPTION
**Pull request by zARODz11z**
### What does this PR do?
Sets http.status_code as a default tag on inferred spans.
Screen Shot 2023-03-16 at 2 08 07 PM

Screen Shot 2023-03-16 at 2 02 32 PM

### Motivation
[SLS-3099
](https://datadoghq.atlassian.net/browse/SLS-3099)

Which originated from [SLES-1059](https://datadoghq.atlassian.net/browse/SLES-1059) where we came to the conclusion we needed to add http.status_code as a default span tag to apigateway inferred spans. This new logic adds http.status_code to all inferred spans.

### Describe how to test/QA your changes
Manually tested in sls sandbox environment, by publishing a version of the serverless agent (extension) and utilizing it as a lambda layer in the following application: API Gateway->Java AWS Lambda function. The results are captured in the screenshots above. [Dogfluence docs explaining my manual testing process](https://datadoghq.atlassian.net/wiki/spaces/SLS/pages/2723611380/Lambda+Extension#Manual-Testing)

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

[SLES-1059]: https://datadoghq.atlassian.net/browse/SLES-1059?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: pkg/serverless/trace/inferredspan/inferred_span.go [hunk 1]
```diff
@@ -171,5 +171,8 @@ func IsInferredSpansEnabled() bool {
 // AddTagToInferredSpan is used to add new tags to the inferred span in
 // inferredSpan.Span.Meta[]. Should be used before completing an inferred span.
 func (inferredSpan *InferredSpan) AddTagToInferredSpan(key string, value string) {
+	if inferredSpan.Span.Meta == nil {
+		inferredSpan.Span.Meta = make(map[string]string)
+	}
```
Comment (zARODz11z) [inline pkg/serverless/trace/inferredspan/inferred_span.go:176]: I was getting a panic when running this test locally

**Comment by zARODz11z** [comment]
Regarding sls integration tests
https://github.com/DataDog/datadog-agent/pull/16054#issuecomment-1474391582

File: pkg/serverless/trace/inferredspan/inferred_span.go (hunk 1 above)
Comment (IvanTopolcic) [reply pkg/serverless/trace/inferredspan/inferred_span.go:176]: Does it make sense to move this earlier on where we create the inferred span?

Comment (zARODz11z) [reply pkg/serverless/trace/inferredspan/inferred_span.go:176]: @IvanTopolcic I ran through my test with a debugger and tried putting this nil check in other spots such as `generateInferredSpan` https://github.com/DataDog/datadog-agent/blob/8c3be8a7aed4d855605a7a2f70b6b25f355f8063/pkg/serverless/trace/inferredspan/inferred_span.go#L156

but it seems my test is hitting some other code path, as we make an[ inferredspan slice ](https://github.com/DataDog/datadog-agent/pull/16156/files#diff-96b011b6370ccc9a3d01c42f36ee2bdc78f78b0144aaf5f30b6dae884021b62bR224) manually in the test then call OnInvokeEnd and thus I think it is at a good spot for the nil check.

File: pkg/serverless/invocationlifecycle/lifecycle.go [hunk 2]
```diff
@@ -229,6 +229,7 @@ func (lp *LifecycleProcessor) OnInvokeEnd(endDetails *InvocationEndDetails) {
 					lp.requestHandler.inferredSpans[1].CompleteInferredSpan(lp.ProcessTrace, lp.getInferredSpanStart(), endDetails.IsError, lp.GetExecutionInfo().TraceID, lp.GetExecutionInfo().SamplingPriority)
 					log.Debug("[lifecycle] The secondary inferred span attributes are %v", lp.requestHandler.inferredSpans[1])
 				}
+				lp.GetInferredSpan().AddTagToInferredSpan("http.status_code", statusCode)
```
Comment (agocs) [inline pkg/serverless/invocationlifecycle/lifecycle.go:232]: Is `statusCode` guaranteed to exist?

**Review by agocs — APPROVED** [review]

File: pkg/serverless/invocationlifecycle/lifecycle.go (hunk 2 above)
Comment (zARODz11z) [reply pkg/serverless/invocationlifecycle/lifecycle.go:232]: statusCode is a string and in go it defaults to an empty string if not initialized

Screen Shot 2023-04-27 at 1 07 35 PM

Screen Shot 2023-04-27 at 1 12 56 PM

**Pull request merged**